base/common/src/main/java/org/dogtagpki/nss/NSSExtensionGenerator.java

```diff
@@ -545,11 +543,36 @@ public Extension createGenericExtension(String oid) throws Exception {
         return new Extension(new ObjectIdentifier(oid), critical, extValue);
     }
 
+    /**
+     * Create extensions.
+     */
     public Extensions createExtensions() throws Exception {
-        return createExtensions(null, null);
+        return createExtensions(null, null, null);
```

Comment:
This is getting on for wanting a builder or factory constructor, but out of scope for this.

Reply:
The current code is sort of a stateless builder which can be shared by multiple threads. If we use a stateful builder (where extensions can be added incrementally) each thread will need its own builder instance. There might be a performance implication, but it might be negligible too, so I don't really have any objection to that.